cmd/gsw_service.go

```
@@ -115,6 +115,8 @@ func readConfig() *viper.Viper {
 	config := viper.New()
 	configFilepath := flag.String("c", "gsw_service", "name of config file")
 	flag.Parse()
+	config.SetDefault("database_port_number", 8089)
+	config.SetDefault("database_host_name", "localhost")
```

Blast these. I assume theres an error that gets thrown if the key isnt found. The user should have this set which is done in the default config

---

👍

---

if nothing is found it does not throw an error, do you want it to? and if it doesn't find anything what do you want the program to do?

---

Yea lets throw an error. By default it wont error if we give it this default file, so that means the user would've had to make an incorrect modification. Debating on the last part. Just exit early i would say now is fine since we do probably want to log our data no matter what, but in the future we configure to allow disabling it

---

so for now, you want it to error if it doesn't detect a database_port_number and database_host_name, and later on you want a setting where you can turn off sending data to the database?